includes/wikia/models/NavigationModel.class.php

```diff
@@ -218,6 +222,24 @@ function () use ( $type, $source, $maxChildrenAtLevel ) {
 		return $menuData;
 	}
 
+	public function getTreeFromText( string $wikiText ): array {
+		$this->setShouldTranslateContent( false );
+		$menuData = [];
+
+		$this->menuNodes = $this->parseText( $wikiText, [
+			$this->wg->maxLevelOneNavElements,
+			$this->wg->maxLevelTwoNavElements,
+			$this->wg->maxLevelThreeNavElements,
+		], true );
+
+		foreach ( $this->menuNodes[0][self::CHILDREN] as $id ) {
+			$menuData[] = $this->recursiveConvertMenuNodeToArray( $id );
+		}
+		$this->setShouldTranslateContent( true );
+
+		return $menuData;
+	}
+
 	/*
 	 * TODO we should refactor whole model when we remove Oasis
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)